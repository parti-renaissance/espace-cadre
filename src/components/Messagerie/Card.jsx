import PropTypes from 'prop-types'
import { Box, Typography } from '@mui/material'
import LockOutlinedIcon from '@mui/icons-material/LockOutlined'
import { generatePath, useNavigate } from 'react-router'
import noImage from '~/assets/no-image.png'
import { paths as messageriePaths } from '~/components/Messagerie/shared/paths'
import Button from '~/ui/Button/Button'
import UICard from '~/ui/Card'

const Card = ({ template }) => {
  const navigate = useNavigate()
  const handleClick = () => {
    navigate(generatePath(`${messageriePaths.create}?templateId=${template.uuid}`))
  }

  return (
    <UICard
      rootProps={{ sx: { p: 0, border: '1px solid', borderColor: 'colors.gray.200', overflow: 'hidden' } }}
      header={
        <Box sx={{ position: 'relative' }}>
          <Box
            sx={{
              height: '8rem',
              bgcolor: 'colors.gray.50',
              display: 'flex',
              alignItems: 'center',
              justifyContent: 'center',
            }}
          >
            <img src={noImage} height="75" width="75" alt="no image" />
          </Box>
          {template.from_admin && (
            <Box
              sx={{
                position: 'absolute',
                top: '10px',
                right: '10px',
                width: '32px',
                height: '32px',
                borderRadius: '4px',
                bgcolor: 'inactiveLabel',
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
              }}
            >
              <LockOutlinedIcon sx={{ color: 'form.error.color', fontSize: '16px' }} />
            </Box>
          )}
        </Box>
      }
      content={
        <Box sx={{ p: 1.5 }}>
          <Typography component="h2" sx={{ fontWeight: '500', color: 'colors.gray.900', fontSize: '16px' }}>
            {template.label}
          </Typography>
        </Box>
      }
      actions={
        <Box sx={{ display: 'flex', alignItems: 'center', p: 1.5 }} className="space-x-2">
          <Button onClick={handleClick} rootProps={{ sx: { fontSize: '12px' } }} isMainButton>
            Utiliser
          </Button>
        </Box>
      }
    />
  )
}

export default Card

Card.propTypes = {
  template: PropTypes.object,
  onDropTemplate: PropTypes.func,
}
