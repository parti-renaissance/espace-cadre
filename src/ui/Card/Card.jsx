import PropTypes from 'prop-types'
import { Card as MuiCard, CardContent } from '@mui/material'
import { styled } from '@mui/system'

const StyledCard = styled(MuiCard)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-between',
  border: '1px solid #eee',
})

const StyledActions = styled('div')({
  display: 'flex',
  width: '100%',
  padding: '16px',
  paddingTop: '0',
  paddingBottom: '8px',
  justifyContent: 'space-between',
  flexDirection: 'column',
})

const UICard = ({
  header,
  content,
  actions,
  rootProps = {},
  headerProps = {},
  contentProps = {},
  actionsProps = {},
}) => (
  <StyledCard {...rootProps} data-cy="ui-card">
    {header && <CardContent {...headerProps}>{header}</CardContent>}
    {content && (
      <CardContent
        sx={
          header
            ? {
                pt: '0',
              }
            : {}
        }
        {...contentProps}
      >
        {content}
      </CardContent>
    )}
    {actions && <StyledActions {...actionsProps}>{actions}</StyledActions>}
  </StyledCard>
)

UICard.propTypes = {
  header: PropTypes.node,
  content: PropTypes.node,
  actions: PropTypes.node,
  rootProps: PropTypes.object,
  headerProps: PropTypes.object,
  contentProps: PropTypes.object,
  actionsProps: PropTypes.object,
}

export default UICard
