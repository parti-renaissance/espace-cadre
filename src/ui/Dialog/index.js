export { default } from './Dialog'
export { ModalForm } from './ModalForm'
