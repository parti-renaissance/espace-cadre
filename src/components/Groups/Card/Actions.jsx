import { CtaButton } from '~/ui/Card'
import { styled } from '@mui/system'
import { generatePath, useNavigate } from 'react-router-dom'
import paths from '../shared/paths'
import PropTypes from 'prop-types'
import DotsMenu, { DotsMenuItem } from '~/ui/Card/Menu/DotsMenu'

const HorizontalContainer = styled('div')`
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: space-between;
`

const messages = {
  see: 'Voir',
  edit: 'Modifier',
  delete: 'Supprimer',
}

const Actions = ({ groupId, onEdit, onDelete, isDeletable }) => {
  const navigate = useNavigate()
  const handleClick = () => {
    navigate(generatePath(`:groupId/${paths.update}`, { groupId }))
  }

  return (
    <HorizontalContainer>
      <CtaButton onClick={handleClick}>{messages.see}</CtaButton>
      <DotsMenu>
        <DotsMenuItem onClick={onEdit}>{messages.edit}</DotsMenuItem>
        {isDeletable && <DotsMenuItem onClick={onDelete}>{messages.delete}</DotsMenuItem>}
      </DotsMenu>
    </HorizontalContainer>
  )
}

export default Actions

Actions.propTypes = {
  groupId: PropTypes.string.isRequired,
  onEdit: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  isDeletable: PropTypes.bool.isRequired,
}
