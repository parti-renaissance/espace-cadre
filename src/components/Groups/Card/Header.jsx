import pluralize from '~/components/shared/pluralize/pluralize'
import PropTypes from 'prop-types'
import { UIChip } from '~/ui/Card'
import { styled } from '@mui/system'

const HorizontalContainer = styled('div')`
  display: flex;
  flex: 1;
  align-items: center;
`

const messages = {
  member: 'militant',
}

const Header = ({ groupCount }) => (
  <HorizontalContainer>
    <UIChip
      color="gray700"
      bgcolor="rgba(55, 65, 81, 0.08)"
      label={`${groupCount} ${pluralize(groupCount, messages.member)}`}
    />
  </HorizontalContainer>
)

Header.propTypes = {
  groupCount: PropTypes.number.isRequired,
}

export default Header
