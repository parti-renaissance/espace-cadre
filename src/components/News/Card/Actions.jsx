import { CtaButton } from '~/ui/Card'
import { styled } from '@mui/system'
import VisibilityOffIcon from '@mui/icons-material/VisibilityOff'
import EditIcon from '@mui/icons-material/EditRounded'
import PropTypes from 'prop-types'
import DotsMenu, { DotsMenuItem } from '~/ui/Card/Menu/DotsMenu'

const HorizontalContainer = styled('div')`
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: space-between;
`

const messages = {
  see: 'Voir',
  published: 'Publier',
  unPublished: 'DÃ©publier',
  edit: 'Modifier',
}

const Actions = ({ status, toggleStatus, handleEdit, statusLoader = false, onView }) => (
  <HorizontalContainer>
    <CtaButton onClick={onView}>{messages.see}</CtaButton>
    <DotsMenu>
      <DotsMenuItem onClick={toggleStatus} loader={statusLoader}>
        <VisibilityOffIcon sx={{ mr: 1, fontSize: '12px' }} />
        {status ? messages.unPublished : messages.published}
      </DotsMenuItem>
      <DotsMenuItem onClick={handleEdit}>
        <EditIcon sx={{ mr: 1, fontSize: '12px' }} />
        {messages.edit}
      </DotsMenuItem>
    </DotsMenu>
  </HorizontalContainer>
)

export default Actions

Actions.propTypes = {
  status: PropTypes.bool.isRequired,
  toggleStatus: PropTypes.func.isRequired,
  statusLoader: PropTypes.bool,
  onView: PropTypes.func.isRequired,
  handleEdit: PropTypes.func.isRequired,
}
