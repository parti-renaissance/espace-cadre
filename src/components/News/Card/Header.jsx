import PropTypes from 'prop-types'
import NotificationsActiveRoundedIcon from '@mui/icons-material/NotificationsActiveRounded'
import NotificationsOffRoundedIcon from '@mui/icons-material/NotificationsOffRounded'
import { UIChip } from '~/ui/Card'
import { styled } from '@mui/system'

const HorizontalContainer = styled('div')`
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: space-between;
`
const IconContainer = styled('div')`
  display: flex;
  flex: 1;
  align-items: center;
`

const NotificationsOnIcon = styled(NotificationsActiveRoundedIcon)(
  ({ theme }) => `
  font-size: 16px;
  font-weight: 500;
  border-radius: 19px;
  padding: ${theme.spacing(0.25)};
  border-color: ${theme.palette.gray100};
  border: ${`1px solid ${theme.palette.gray200}`};
  margin: ${theme.spacing(0.25, 1, 0, 1)};
`
)

const NotificationsOffIcon = styled(NotificationsOffRoundedIcon)(
  ({ theme }) => `
  font-size: 16px;
  font-weight: 500;
  border-radius: 19px;
  padding: ${theme.spacing(0.25)};
  border-color: ${theme.palette.gray100};
  border: 1px solid ${theme.palette.gray200};
  margin: ${theme.spacing(0.25, 0, 0, 1)};
`
)

const messages = {
  published: 'Publiée',
  unpublished: 'Dépubliée',
}

const Header = ({ status, withNotification }) => (
  <HorizontalContainer>
    <IconContainer>
      <UIChip
        color={status ? 'teal700' : 'gray700'}
        bgcolor={status ? 'activeLabel' : 'inactiveLabel'}
        label={status ? messages.published : messages.unpublished}
      />
      {withNotification ? <NotificationsOnIcon sx={{ mr: 1 }} /> : <NotificationsOffIcon sx={{ mr: 1 }} />}
    </IconContainer>
  </HorizontalContainer>
)

Header.propTypes = {
  status: PropTypes.bool.isRequired,
  withNotification: PropTypes.bool.isRequired,
}

export default Header
