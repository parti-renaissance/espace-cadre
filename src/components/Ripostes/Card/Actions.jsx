import { CtaButton } from '~/ui/Card'
import { styled } from '@mui/system'
import PropTypes from 'prop-types'
import DotsMenu, { DotsMenuItem } from '~/ui/Card/Menu/DotsMenu'

const HorizontalContainer = styled('div')`
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: space-between;
`

const messages = {
  activate: 'Activer',
  deactivate: 'Désactiver',
  edit: 'Éditer',
}

const Actions = ({ status, toggleStatus, loader = false, onEdit }) => (
  <HorizontalContainer>
    <CtaButton onClick={onEdit}>{messages.edit}</CtaButton>
    <DotsMenu>
      <DotsMenuItem onClick={toggleStatus} loader={loader}>
        {status ? messages.deactivate : messages.activate}
      </DotsMenuItem>
    </DotsMenu>
  </HorizontalContainer>
)

export default Actions

Actions.propTypes = {
  status: PropTypes.bool.isRequired,
  toggleStatus: PropTypes.func.isRequired,
  loader: PropTypes.bool,
  onEdit: PropTypes.func.isRequired,
}
