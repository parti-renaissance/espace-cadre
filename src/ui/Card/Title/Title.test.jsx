import { render } from '@testing-library/react'
import { Title } from 'ui/Card'

vi.mock('@mui/system', () => ({
  styled: c => () => c,
}))
vi.mock('@mui/icons-material/Person', () => ({ default: () => <div className="mock_PersonIcon" /> }))
vi.mock('@mui/icons-material/AccessTime', () => ({ default: () => <div className="mock_AccessTimeIcon" /> }))

vi.mock('@mui/material', async importOriginal => {
  const actual = await importOriginal()
  return {
    ...actual,
    Typography: ({ children }) => <div className="mui-typography-mock">{children}</div>,
    // your mocked methods
  }
})

describe('Title', () => {
  it('displays a Title', () => {
    const { container } = render(
      <Title subject="subject" author="author" dateTime={new Date(Date.UTC(2022, 2, 17, 8))}></Title>
    )
    expect(container).toMatchSnapshot()
  })
})
