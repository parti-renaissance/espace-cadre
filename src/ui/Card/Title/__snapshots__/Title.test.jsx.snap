// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Title > displays a Title 1`] = `
<div>
  <div>
    <div
      class="mui-typography-mock"
    >
      subject
    </div>
    <div>
      <div
        class="mui-typography-mock"
      >
        author
      </div>
    </div>
    <div>
      <div
        class="mui-typography-mock"
      >
        Le 
        17/03/2022
         à 
        09:00
      </div>
    </div>
  </div>
</div>
`;

exports[`Title displays a Title 1`] = `
<div>
  <div>
    <div
      class="mui-typography-mock"
    >
      subject
    </div>
    <div>
      <div
        class="mock_PersonIcon"
      />
      <div
        class="mui-typography-mock"
      >
        author
      </div>
    </div>
    <div>
      <div
        class="mock_AccessTimeIcon"
      />
      <div
        class="mui-typography-mock"
      >
        Le 
        17/03/2022
         à 
        09:00
      </div>
    </div>
  </div>
</div>
`;
